src/lib/utils/validators.ts

```diff
@@ -10,6 +10,37 @@ import type {
 } from '$lib/types/journey';
 import { to_section_list } from './converters';
 
+/**
+ * Establish whether a component's value is valid, trusting the component's judgement if provided
+ */
+export function component_valid(component: InputComponent, state: StateStoreType): boolean {
+	// (ineligible) - component has no identifier, it must be a display component only, we have no reason to validate
+	if (!component.id) return true;
+
+	// (passed|failed) - input component has been attempted and has a validation status, trust this value
+	if (state[component.id]?.valid != null) return state[component.id]?.valid;
+
+	// (skipped) - component is optional and has been skipped, don't need to validate
+	if (!component.required ?? false) return true;
+
+	// (hidden) - component is hidden due to dependency on another component, don't fail validation
+	if (
+		component.dependsupon &&
+		state[component.dependsupon.id]?.value != component.dependsupon.value
+	)
+		return true;
```

Comment:
Add some { } - difficult to read.. I misread it 4 times

Reply:
have pushed this up